### ISSUE TITLE
Add CombinedTM / ZeroShotTM (contextualized neural topic models)

### ISSUE THREAD

**Issue by nealcaren** [nealcaren/topica#172]
## Summary

Add **CombinedTM** and **ZeroShotTM** (Bianchi et al. 2021), the contextualized neural topic models. These are the most widely used modern NTMs and the single biggest gap relative to the Wu, Nguyen & Luu (2024) survey [*A Survey on Neural Topic Models*](https://arxiv.org/abs/2401.15351) (§ PLM-based / embedding NTMs). They also supply the survey's cross-lingual-via-PLM story.

## Why this is cheap

These are **ProdLDA with a different encoder input** — nothing downstream changes. The existing machine in `src/prodlda.rs` (encoder `V → hidden → hidden → (μ, logvar)`, batchnorm heads, reparameterize, product-of-experts decoder `β (K×V)`, Laplace logistic-normal prior, FD-checked hand-coded backward) is reused verbatim.

- **CombinedTM**: encoder input = normalized BoW **concatenated with** an SBERT document embedding. Layer-1 widens from `V` (sparse) to `V + E` (sparse BoW block + dense `E` block). Decoder, KL, loss, backward untouched.
- **ZeroShotTM**: encoder input = **only** the document embedding (`E → hidden`, dense); decoder still reconstructs BoW. Inference reads only the embedding, so a test document embedded with a multilingual encoder yields topics zero-shot — cross-lingual transfer with no refit.

## Proposed API

Mirror `ETM.fit`, which already takes caller-supplied vectors + aligned vocabulary (`src/python.rs:11146`). You bring the embeddings (sentence-transformers / API / ollama), consistent with the rest of the embedding-based models.

```python
m = CombinedTM(num_topics=20, hidden_size=100, seed=42)
m.fit(corpus, doc_embeddings, iters=200)        # doc_embeddings: (n_docs, E)

m = ZeroShotTM(num_topics=20, seed=42)
m.fit(en_corpus, en_embeddings)
m.transform(es_docs, es_embeddings)             # Spanish topics, no refit
```

## Implementation notes

- New Rust is limited to a widened layer-1 input and a `doc_embeddings` plumb-through; reuse `Weights`/`BatchNorm`/`Adam`/`Optim` from `prodlda.rs`.
- Keep the FD gradient-check test pattern (`prodlda.rs::batch_gradients_match_fd`) for the augmented encoder.
- Preserve determinism (fixed seed + thread count) and the no-PyTorch guarantee.
- Validate against the `contextualized-topic-models` reference package (planted-topic recovery + coherence parity).

Ref: Bianchi, Terragni & Hovy 2021 (CombinedTM, ACL); Bianchi et al. 2021 (ZeroShotTM, EACL).

**Comment by nealcaren** [comment]
Design note — new class vs. `ProdLDA` option:

Internally these *are* ProdLDA with a widened encoder input, so by a strict "same generative story" test they could be exposed as `ProdLDA(context_embeddings=...)`. Two things argue for keeping them as their own classes:

1. **Different `fit` signature** — they require document embeddings at fit time, unlike plain `ProdLDA`. This mirrors how `ETM` is its own class because it needs `word_embeddings`/`vocabulary`.
2. **Discoverability** — heavily cited under their own names; users will reach for `from topica import CombinedTM`.

This is consistent with `BERTopic`/`Top2Vec`/`ETM`/`FASTopic` all being distinct classes over shared embedding infra. **Recommendation: separate classes that call the prodlda core.** Of the six survey-gap issues, this is the one most cleanly collapsible into an existing class if we ever want to minimize the public class count — noting that here so the decision is explicit.

(Contrast: #174 contrastive and #176 alternative priors are deliberately *flags* on existing models, not new classes, because they change only the objective/prior on the same generative story + same `fit` shape — matching the `inference=`/`sampler=` precedent.)

**Comment by nealcaren** [comment]
Sequencing: land the model registry (#182) first so this slots into the new taxonomy (group: embedding-based; tags: cross-lingual for ZeroShotTM) rather than lengthening the flat model list.
